### DISCUSSION TITLE
a new version of the dashboard

### DISCUSSION THREAD

**Discussion by NICEXAI** [apolloconfig/apollo#4048] [Ideas]
I think using a technology stack that everyone is familiar with allows more people to participate.
I want to redevelop a new version of the dashboard using react + hook + ts or vue3 + ts.
Also I want to redesign the pages and interactions.

I would like to hear your opinions.

**Comment by nobodyiam** [comment]
That's a great idea! But it definitely requires a huge amount of work.
Currently, apollo-portal is written with angular js which is at least understandable to me 😀.
I'm not quite familiar with react + hook + ts or vue3 + ts but I think this is a topic worth discussing. cc @apolloconfig/committers

**Reply by NICEXAI** [reply]
> That's a great idea! But it definitely requires a huge amount of work. Currently, apollo-portal is written with angular js which is at least understandable to me 😀. I'm not quite familiar with react + hook + ts or vue3 + ts but I think this is a topic worth discussing. cc @apolloconfig/committers

Yes, there is indeed a lot that needs to be done. I should be able to help with a few things.

**Reply by NICEXAI** [reply]
The framework is just a tool, even if you change to another framework, I believe you can spend very little time to get familiar with it.

**Reply by NICEXAI** [reply]
In the vast majority of cases, redevelopment without burden is far more efficient than refactoring.

**Comment by nobodyiam** [comment]
There is also discussion in #4051 which suggests upgrading AngularJs to Angular. I think we could merge these discussions.

**Comment by kezhenxu94** [comment]
+1 to vuejs + ts

**Comment by nisiyong** [comment]
+1 to vuejs + ts
